.github/workflows/release.yml

```diff
@@ -0,0 +1,52 @@
+---
+name: "TER deployment"
+
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+
+jobs:
+  # The actual TER upload job
+  ter-release:
+    name: "TYPO3 TER release"
+
+    # Run only if it's a tag in general (as regex is not possible here)
+    if: startsWith(github.ref, 'refs/tags/')
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        php-versions: [7.4]
+
+    env:
+      TYPO3_EXTENSION_KEY: tea
+      TYPO3_USER: ${{ secrets.TYPO3_USER }}
+      TYPO3_PASSWORD: ${{ secrets.TYPO3_PASSWORD }}
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          path: ${{ env.TYPO3_EXTENSION_KEY }}
+
+      - uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-versions }}
+          extensions: intl, mbstring, xml, soap, zip, curl
+
+      - name: Install TYPO3 TER client
+
+        run: composer global require namelesscoder/typo3-repository-client
+          "^2.0" --prefer-dist --no-progress --no-suggest
+
+      - name: Remove .git folder
+        run: rm -rf ./$TYPO3_EXTENSION_KEY/.git
+
+      - name: List extension folder
+        run: ls -liAsh ./$TYPO3_EXTENSION_KEY
+
+      - name: Upload EXT:${{ env.TYPO3_EXTENSION_KEY }}
```

Comment:
Is there any way we could read this from the composer.json (where the key already should be listed)?